lib/autoload.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+class ClassLoader {
+    private $searchPath = [];
+
+    /**
+     * Register path in which to look for a class.
+     */
+    public function registerPath($path) {
+        if (is_string($path)) {
+            $this->searchPath[] = $path;
+        }
+        if (is_array($path)) {
+            array_push($this->searchPath, ...$path);
+        }
+    }
+
+    /**
+     * Load class file if found.
+     */
+    public function loadClass($class)
+    {
+        if ($file = $this->findFile($class)) {
+            require $file;
+        }
+    }
+
+    /**
+     * Find the file containing the class definition.
+     */
+    public function findFile($class) {
+        // This match most of classes directly
+        foreach ($this->searchPath as $path) {
+            $file = $path . DIRECTORY_SEPARATOR . str_replace(['\\', '_'], DIRECTORY_SEPARATOR, $class) . '.php';
+            if (file_exists($file)) {
+                return $file;
+            }
+        }
+
+        if (false === strpos($class, 'FreshRSS_')) {
+            return;
+        }
+
+        // This match FRSS classes
+        $freshrssClass = str_replace(['FreshRSS_', '_'], '', $class);
+        foreach ($this->searchPath as $path) {
+            foreach (glob($path . DIRECTORY_SEPARATOR . '*', GLOB_ONLYDIR) as $subPath) {
+                $file = $subPath . DIRECTORY_SEPARATOR . $freshrssClass . '.php';
+                if (file_exists($file)) {
```

Comment:
On a second read, I am a bit worried about nested loops and multiple IO operations to load a file for which we are supposed to know the exact path

Reply:
I removed the loops by adding manually the path during configuration.